### DISCUSSION TITLE
css-loader hash collisions for localIdentName: [contenthash:base64:1]

### DISCUSSION THREAD

**Discussion by kn9ka** [webpack/webpack#14799] [Q&A]
Hello,
I ran into a hash collision problem when used localIdentName: [contenthash:base64:4]. We have already fixed it, but there were some questions :)
Is it only way to generate hash name for each separated chunk? 
Why css-loader cannot generate hash names for classnames before chunk separate or use some unique(e.g. timestamp) prefix for generate hash?

reproduce: generate more then 100 classnames and use localIdentName: [contenthash:base64:1] for css-loader options.

**Comment by alexander-akait** [comment]
Can you provide reproducible example? Should be no collisions

**Reply by kn9ka** [reply]
It is rarely bug and hard to reproduce without large codebase each build with unique hash can generate overlapping names.
I can show you two chunks of builded code with overlapped names.

**Reply by alexander-akait** [reply]
Yep, please provide example, also can you provide options for css-loader?

**Reply by kn9ka** [reply]
I am sorry but my NDA does not allow me to provide  chunks only screen.
![telegram-cloud-photo-size-2-5251622509466466658-x](https://user-images.githubusercontent.com/13977991/143015368-e04f04ca-b5b0-4387-ad8e-b632ebbd1784.jpg)

css-loader options:

```
  {
    loader: 'css-loader',
    options: {
      sourceMap: false,
      modules: {
        auto: /\.module\.(css|scss)$/,
        localIdentName: '[contenthash:base64:4]',
      },
      importLoaders: 1,
    },
  },
```

**Reply by alexander-akait** [reply]
Do you use something else except `css-loader`?

**Reply by alexander-akait** [reply]
Also `4` is very short... Can yo try to setup it more and check again?

**Reply by kn9ka** [reply]
Check my initial post please :)
We have already fixed it by increased to hash:20
btw, hash:4 means 16 million combination of base64(64^4) symbols, why was classname same?

**Reply by alexander-akait** [reply]
hm, agree, maybe edge case... what is `css-loader` version?

**Reply by alexander-akait** [reply]
Also we merged https://github.com/webpack-contrib/css-loader/pull/1382, maybe something wrong here, but it should decrease collisions

**Reply by kn9ka** [reply]
```
"css-loader": "5.2.6",
```

**Reply by alexander-akait** [reply]
Can you update to `v6`? There are were a lot of fixes...

**Reply by kn9ka** [reply]
sure :)
